### PR TITLE
some pep8 cleanup in four crystal files

### PR DESCRIPTION
**Pull request by fchapoton**
a somewhat large scale pep8 cleanup in the 4 modified files in the crystals folder

also a few minor code enhancements

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/crystals/alcove_path.py [hunk 1]
```diff
@@ -1040,12 +1040,13 @@ def f(self, i):
         #
         # otherwise if m_index - 1 > 0 then (C2) is enough
 
-        if ( (not finite_cartan_type or i != 0) and M > 0  # alpha_i is a simple root
-           ) or KR_test :# KR case
+        if ((not finite_cartan_type or i != 0) and M > 0  # alpha_i is a simple root
+            ) or KR_test:  # KR case
```
Comment (dcoudert) [inline src/sage/combinat/crystals/alcove_path.py]: I don't get the alignment here. the `)` is closing `((not finite_cartan_type or i != 0) and M > 0'.
In fact, it's weird to have this parenthesis on a new line...

File: src/sage/combinat/crystals/generalized_young_walls.py [hunk 2]
```diff
@@ -394,7 +394,7 @@ def number_of_parts(self):
                 new.pop(i)
             elif r[0] == n and len(r) % (n+1) == 0:
                 for j in range(n+1):
-                    temp = [k % (n+1) for k in range(j+len(r)/(n+1)-1,j-1,-1)]
+                    temp = [k % (n+1) for k in range(j+len(r)/(n+1)-1, j-1, -1)]
```
Comment (dcoudert) [inline src/sage/combinat/crystals/generalized_young_walls.py]: you could add spaces around operators (+, -, /). You did it elsewhere

File: src/sage/combinat/crystals/generalized_young_walls.py [hunk 3]
```diff
@@ -514,7 +514,7 @@ def latex_large(self):
             s += "\\emptyset"
         else:
             s += "\\begin{tikzpicture}[baseline=5,scale=.45] \n \\foreach \\x [count=\\s from 0] in \n"
-            s += "{" + ','.join("{" + ','.join( str(i) for i in r ) + "}" for r in self.data ) + "} \n"
+            s += "{" + ','.join("{" + ','.join(str(i) for i in r) + "}" for r in self.data) + "} \n"
             s += "{\\foreach \\y [count=\\t from 0] in \\x {  \\node[font=\\scriptsize] at (-\\t,\\s) {$\\y$}; \n \\draw (-\\t+.5,\\s+.5) to (-\\t-.5,\\s+.5); \n \\draw (-\\t+.5,\\s-.5) to (-\\t-.5,\\s-.5); \n \\draw (-\\t-.5,\\s-.5) to (-\\t-.5,\\s+.5);  } \n \\draw[-,thick] (.5,\\s+1) to (.5,-.5) to (-\\t-1,-.5); } \n \\end{tikzpicture} \n"
```
Comment (dcoudert) [inline src/sage/combinat/crystals/generalized_young_walls.py:520]: this line could be split

File: src/sage/combinat/crystals/generalized_young_walls.py [hunk 4]
```diff
@@ -554,13 +554,10 @@ def weight(self, root_lattice=False):
...
             -2*alpha[0] - 3*alpha[1] - 5*alpha[2] - 3*alpha[3]
         """
-        W = []
         E = self.cartan_type().root_system().weight_lattice(extended=True)
         L = self.cartan_type().root_system().root_lattice()
         alpha = L.simple_roots()
-        for r in self.data:
-            for i in r:
-                W.append(-1*alpha[i])
+        W = [-1*alpha[i] for r in self.data for i in r]
```
Comment (dcoudert) [inline src/sage/combinat/crystals/generalized_young_walls.py]: why using `-1*alpha[i]` and not using `-alpha[i]` ?

File: src/sage/combinat/crystals/kirillov_reshetikhin.py [hunk 5]
```diff
@@ -2577,15 +2580,15 @@ def epsilon0(self):
 
         EXAMPLES::
 
-            sage: K=crystals.KirillovReshetikhin(['C',3,1],3,1)
+            sage: K = crystals.KirillovReshetikhin(['C',3,1],3,1)
```
Comment (dcoudert) [inline src/sage/combinat/crystals/kirillov_reshetikhin.py]: spaces missing : `['C', 3, 1], 3, 1`

**Review by dcoudert — CHANGES_REQUESTED** [review]

File: src/sage/combinat/crystals/generalized_young_walls.py (hunk 4 above)
Comment (fchapoton) [reply src/sage/combinat/crystals/generalized_young_walls.py]: just keeping what is written as there may be a good reason to use this

**Review by dcoudert — APPROVED** [review]
LGTM.

File: src/sage/combinat/crystals/generalized_young_walls.py (hunk 4 above)
Comment (dcoudert) [reply src/sage/combinat/crystals/generalized_young_walls.py]: You are right, it's safer.

**Pull request merged**